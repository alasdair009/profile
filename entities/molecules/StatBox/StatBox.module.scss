@use "../../../app/styles/dimensions";

.root {
  align-items: center;
  aspect-ratio: 1;
  display: flex;
  flex-direction: column;
  justify-content: center;
  width: dimensions.getSize(s96);
}
