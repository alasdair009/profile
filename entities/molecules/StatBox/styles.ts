"use client";

import styled from "styled-components";
import { sizes } from "@/entities";

export const Root = styled.article`
  align-items: center;
  aspect-ratio: 1;
  display: flex;
  flex-direction: column;
  justify-content: center;
  width: ${sizes.s96.rem};
`;
