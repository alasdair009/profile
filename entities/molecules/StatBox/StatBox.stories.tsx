import { Meta, StoryObj } from "@storybook/react";
import { StatBox } from "./StatBox";

const meta: Meta<typeof StatBox> = {
  component: StatBox,
  args: {
    heading: "Record",
    name: "name of holder",
    value: "10",
  },
};
export default meta;

export const Default: StoryObj<typeof StatBox> = {};
