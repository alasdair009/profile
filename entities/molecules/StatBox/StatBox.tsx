import { Heading, Paragraph, Spacer } from "@/entities";
import { Root } from "./styles";

type StatBoxProps = {
  /**
   * Heading of the record.
   */
  heading: string;
  /**
   * Name of the holder.
   */
  name: string;
  /**
   * Value of the holder.
   */
  value: string;
};

/**
 * Small box that displays a record and value.
 */
export function StatBox({ heading, name, value, ...rest }: StatBoxProps) {
  return (
    <Root data-testid={StatBox.name} {...rest}>
      <Heading level="h4" as="h2" align="center">
        {heading}
      </Heading>
      <Paragraph align="center">{name}</Paragraph>
      <Spacer />
      <Paragraph align="center">{value}</Paragraph>
    </Root>
  );
}
