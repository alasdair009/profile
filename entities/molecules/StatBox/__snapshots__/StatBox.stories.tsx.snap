// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`molecules/StatBox Default play-test 1`] = `
<article data-testid="StatBox"
         class="sc-hDcvty fGIVZb"
>
  <h2 data-testid="Heading"
      class="sc-hfvVTD iPMJRx"
  >
    Record
  </h2>
  <p data-testid="Paragraph"
     class="sc-gLXSEc jKmJic"
  >
    name of holder
  </p>
  <br data-testid="Spacer"
      class="sc-pFPEP zdPYW"
  >
  <p data-testid="Paragraph"
     class="sc-gLXSEc jKmJic"
  >
    10
  </p>
</article>
`;
